contracts/Donations.sol

```diff
@@ -110,18 +112,21 @@ contract Donations is ERC721, AccessControl {
 
     /**
      * Mints an NFT for every donation in @param _params.
-     * The @param _groupId is used to uniquely identify this group of donations.
-     * Ideally, @param _groupId is the hash of the transaction where the yield for the donations was claimed to by the treasury.
+     * The @param _txHash and @param _batchNr uniquely identify a group of donations.
      *
-     * @param _groupId Unique identifier for the group of donations in @param _params.
+     * @param _txHash The hash of the transaction where the yield for these donations was claimed.
+     * @param _batchNr When there are too many donations in a claim, we break them into batches to not reach the gas limit.
      * @param _params Donation params.
      */
-    function mint(bytes32 _groupId, DonationParams[] calldata _params)
-        external
-        onlyRole(WORKER_ROLE)
-    {
+    function mint(
+        bytes32 _txHash,
+        uint256 _batchNr,
+        DonationParams[] calldata _params
+    ) external onlyRole(WORKER_ROLE) {
+        bytes32 groupId = keccak256(abi.encodePacked(_txHash, _batchNr));
+
         require(
-            !processedDonationsGroups[_groupId],
+            !processedDonationsGroups[groupId],
             "Donations: already processed"
         );
```

Comment:
Yes, this is only to make sure that if the worker for some reason tries to submit the same donations twice, we don't allow it. Nothing is stopping a malicious worker from messing up. I don't see any way to do it.

Reply:
Do we store donation info in the backend when user deposit?

Reply:
@gabrielpoca and i cannot get the point from `batchNr` name, what do you think to change variable to `nonce`?.

Reply:
@ryuheimat the information will be on a subgraph. The backend simply reads from there and mints the NFT in here. I'm a bit afraid of calling it `nonce` because that name already has a meaning in this context. The batch number is only necessary because we may have an infinite number of donations when we claim the yield, and we can't mint an NFT for each in one go. We will have to break those into multiple chunks (batches)